controllers/servers/utils.py

```diff
@@ -742,3 +767,21 @@ def validate_modify_volume_group_request(request):
     _validate_minimum_request_fields(request, ["volume_group_id"])
 
     logger.debug("modify volume group validation finished")
+
+
+def get_replication_object_type_and_id_info(request):
+    object_type = servers_settings.VOLUME_TYPE_NAME
+    object_id = request.volume_id
+    object_info = None
+    if object_info:
+        logger.info(object_info)
+        if object_info.HasField(servers_settings.VOLUME_TYPE_NAME):
+            object_id = object_info.volume.volume_id
+            object_type = servers_settings.VOLUME_TYPE_NAME
+        elif object_info.HasField(servers_settings.VOLUME_GROUP_TYPE_NAME):
+            object_id = object_info.volume.volume_group_id
+            object_type = servers_settings.VOLUME_GROUP_TYPE_NAME
+        else:
+            return object_type, object_id
```

Comment:
you set object_info as None, so it will ever continue to the `if` statement?

Reply:
Yes, I know. Until yesterday we had no object_info inside the request, now I updated spec and will add the real object_info